### PR TITLE
hardhat compatible artifacts generation

### PR DESCRIPTION
**Pull request by Aniket-Engg**

### CODE REVIEW

File: apps/remix-ide/src/app/files/compiler-metadata.js [hunk 1]
```diff
@@ -29,6 +29,28 @@ class CompilerMetadata extends Plugin {
     return joinPath(path, this.innerPath, contractName + '_metadata.json')
   }
 
+  createHardhatArtifacts (compiledContract, provider) {
+    const contract = compiledContract
+    var hhArtifactsFileName = joinPath('artifacts',contract.file, contract.name + '.json')
```
Comment (yann300) [inline apps/remix-ide/src/app/files/compiler-metadata.js:34]: Do you think `_JSONFileName` can be used to process that? also I don't see the `innerPath` being used here.

File: apps/remix-ide/src/app/files/compiler-metadata.js [hunk 2]
```diff
@@ -51,6 +51,8 @@ class CompilerMetadata extends Plugin {
   }
 
   async checkHardhatCache (compiledContract, provider, input, output, versionString) {
+    if (this.fileManager.mode === 'localhost')
+      console.log('Absolute shared folder path', await this.fileManager.sharedFolder())
```
Comment (LianaHus) [inline apps/remix-ide/src/app/files/compiler-metadata.js]: you would need a space.

File: apps/remix-ide/src/app/files/fileManager.js [hunk 3]
```diff
@@ -566,6 +566,13 @@ class FileManager extends Plugin {
     return this._deps.filesProviders.workspace
   }
 
+  async sharedFolder () {
+    if (this.mode === 'localhost') {
+      const provider = this.getProvider('localhost')
+      return await provider.sharedFolder()
+    }
```
Comment (LianaHus) [inline apps/remix-ide/src/app/files/fileManager.js:573]: maybe worth to throw an error or return '' for else

**Review by Hbutlercapone — APPROVED** [review]

**Comment by yann300** [comment]
closing it, @Aniket-Engg just reopen it if needed

**Pull request closed without merging**